src/uct/base/uct_iface.h

```diff
@@ -673,6 +673,33 @@ size_t uct_iov_total_length(const uct_iov_t *iov, size_t iovcnt)
     return total_length;
 }
 
+/**
+ * Fill iovec data structure by data provided in uct_iov_t.
+ * The function avoids copying IOVs with zero length.
+ * @return Number of elements in io_vec[].
+ */
+static UCS_F_ALWAYS_INLINE
+size_t uct_iovec_fill_iov(struct iovec *io_vec, const uct_iov_t *iov,
+                          size_t iovcnt, size_t *total_length)
+{
+    size_t iov_it, io_vec_it = 0;
+
+    *total_length = 0;
+
+    for (iov_it = 0; iov_it < iovcnt; ++iov_it) {
+        io_vec[io_vec_it].iov_len = uct_iov_get_length(&iov[iov_it]);
```

Comment:
maybe cache io_vec[io_vec_it].iov_len in a local variable?

Reply:
done